### PR TITLE
Add display init code for Lilygo TTGO T8 ESP32-S2

### PR DESCRIPTION
**Pull request by m4tk**

### CODE REVIEW

**Comment by netroy** [comment]
i'll test and leave my comments later. thanks for doing this :)

File: ports/esp32s2/boards/lilygo_ttgo_t8_s2_st7789/pins.c [hunk 1]
```diff
@@ -1,11 +1,11 @@
 #include "shared-bindings/board/__init__.h"
+#include "shared-module/displayio/__init__.h"
 
 STATIC const mp_rom_map_elem_t board_global_dict_table[] = {
     { MP_ROM_QSTR(MP_QSTR_IO0), MP_ROM_PTR(&pin_GPIO0) },
     { MP_ROM_QSTR(MP_QSTR_IO1), MP_ROM_PTR(&pin_GPIO1) },
     { MP_ROM_QSTR(MP_QSTR_IO2), MP_ROM_PTR(&pin_GPIO2) },
     { MP_ROM_QSTR(MP_QSTR_IO3), MP_ROM_PTR(&pin_GPIO3) },
-    { MP_ROM_QSTR(MP_QSTR_IO4), MP_ROM_PTR(&pin_GPIO4) },
```
Comment (tannewt) [inline ports/esp32s2/boards/lilygo_ttgo_t8_s2_st7789/pins.c:8]: Is this intentional?

**Review by tannewt — COMMENTED** [review]
Code looks good! Will approve and merge after @netroy tests. Thanks!

File: ports/esp32s2/boards/lilygo_ttgo_t8_s2_st7789/pins.c (hunk 1 above)
Comment (m4tk) [reply ports/esp32s2/boards/lilygo_ttgo_t8_s2_st7789/pins.c:8]: Yes, GPIO4 is the LCD_MISO pin…that‘s why I removed it

**Comment by skieast** [comment]
I don't believe IO4 is connected to MISO on the board. When I was working on the UF2 bootloader I attempted to query the ST7789 using this and couldn't get it to work.  It also is not shown on the schematic on LilyGo Github. 
https://github.com/Xinyuan-LilyGO/LilyGo-T-Display-S2/blob/master/schematic/ESP32_S2-Display.pdf

**Comment by m4tk** [comment]
@skieast thanks for pointing this out. I had just taken this over from the Lily example app :-) Ok, I removed MISO from the code.

**Comment by netroy** [comment]
I was able to replace by `FourWire`/`ST7789` with simply `board.DISPLAY`, and everything worked as expected. :heart:

**Comment by m4tk** [comment]
@netroy thanks for testing this :-)

**Comment by netroy** [comment]
Not sure if this the ticket to discuss this, but I installed the latest `tinyuf2`, to test out [@skieast's PR](https://github.com/adafruit/tinyuf2/pull/67). 
While both these PRs work as expected individually, once I have both installed on the board, I'm seeing some ghosting when using displayio (the slightly dark text at the top and the bottom of the screen).

photo_2021-02-05_15-04-57

is there some kinda of display-reset that I need to do flush out the previous graphics completely? or is that something that needs to happen in this code?

**Comment by skieast** [comment]
> 
> 
> Not sure if this is the ticket to discuss this, but I installed the latest `tinyuf2`, to test out [@skieast's PR](https://github.com/adafruit/tinyuf2/pull/67).
> While both these PRs work as expected individually, once I have both installed on the board, I'm seeing some ghosting when using displayio (the slightly dark text at the top and the bottom of the screen).
> 
> photo_2021-02-05_15-04-57
> 
> is there some kinda of display-reset that I need to do, to flush out the previous graphics completely?
> or is that something that needs to happen in this code?

I'm out most of today but I'm guessing we should probably coordinate the init code in both apps (TinyUF2 and CPY) It's interesting though.  I had tried some test code but didn't notice this.  Thanks,

**Comment by m4tk** [comment]
hmm strange, I also have the tinyuf2 installed…but didn’t experienced such ghosting effects yet…

**Comment by m4tk** [comment]
@netroy can you provide your test script please? Then I will check if this ghosting is also visible on my board.

**Comment by netroy** [comment]
@m4tk the more I try to reproduce this, the weaker the ghosting gets. it's at a point where I can't see it anymore. 
I'll ping you back, if I manage to consistently reproduce it.

File: ports/esp32s2/boards/lilygo_ttgo_t8_s2_st7789/pins.c [hunk 2]
```diff
@@ -41,13 +42,13 @@ STATIC const mp_rom_map_elem_t board_global_dict_table[] = {
     { MP_ROM_QSTR(MP_QSTR_SD_CS), MP_ROM_PTR(&pin_GPIO10) },
 
     // 1.14 inch LCD ST7789
-    { MP_ROM_QSTR(MP_QSTR_LCD_MISO), MP_ROM_PTR(&pin_GPIO4) },
```
Comment (tannewt) [inline ports/esp32s2/boards/lilygo_ttgo_t8_s2_st7789/pins.c:44]: Please leave this. There is a chance folks will want to re-initialize the display themselves. (You can have two names map to the same pin here.)

**Review by tannewt — CHANGES_REQUESTED** [review]
One pin suggestion.

File: ports/esp32s2/boards/lilygo_ttgo_t8_s2_st7789/pins.c (hunk 2 above)
Comment (m4tk) [reply ports/esp32s2/boards/lilygo_ttgo_t8_s2_st7789/pins.c:44]: Does it makes sense if MISO is not connected to IO4? according to @skieast and the schematics on LilyGo repo it isn't connected.

Comment (m4tk) [reply ports/esp32s2/boards/lilygo_ttgo_t8_s2_st7789/pins.c:44]: @skieast but as I'm looking on the schematic now...what about IO37? how can I test if IO37 is MISO? 
image

Comment (skieast) [reply ports/esp32s2/boards/lilygo_ttgo_t8_s2_st7789/pins.c:44]: Unfortunately their labels don't match the actual function.  As their examples don't match the hardware. I spent a few hours trying to read the ST7789 registers with no success, trying all the various pins. 
Looking at he OLED end they label everyything as if it was I2C and not SPI
image

I rang out the various pins using a meter and there is no MISO on the display.  They use the same display on their GD32 board and it more clearly shows the connections.

Comment (skieast) [reply ports/esp32s2/boards/lilygo_ttgo_t8_s2_st7789/pins.c:44]: image
Here are the connections to the same display on the GD32 board

Comment (m4tk) [reply ports/esp32s2/boards/lilygo_ttgo_t8_s2_st7789/pins.c:44]: ok, then I would prefer to leave the code as it is. no MISO pin for the display.

Comment (tannewt) [reply ports/esp32s2/boards/lilygo_ttgo_t8_s2_st7789/pins.c:44]: Thanks for running this down!

**Review by tannewt — APPROVED** [review]
Thank you!

**Pull request merged**